src/huggingface_hub/inference/_client.py

```diff
@@ -1238,6 +1238,66 @@ def text_to_speech(self, text: str, *, model: Optional[str] = None) -> bytes:
         """
         return self.post(json={"inputs": text}, model=model, task="text-to-speech")
 
+    def token_classification(
+        self, text: List[str], *, parameters: Optional[Dict[str, Any]] = None, model: Optional[str] = None
+    ) -> List[ClassificationOutput]:
+        """
+        Perform token classification on the given text.
+        Usually used for sentence parsing, either grammatical, or Named Entity Recognition (NER) to understand keywords contained within text.
+
+        Args:
+            text (`str`):
+                A list of strings to be classified.
+            parameters (`Dict[str, Any]`, *optional*):
+                Additional parameters for the token classification task. Defaults to None. For more details about the available
+                parameters, please refer to [this page](https://huggingface.co/docs/api-inference/detailed_parameters#token-classification-task)
+            model (`str`, *optional*):
+                The model to use for the token classification task. Can be a model ID hosted on the Hugging Face Hub or a URL to
+                a deployed Inference Endpoint. If not provided, the default recommended token classification model will be used.
+                Defaults to None.
+
+        Returns:
+            `List[Dict]`: a list of dictionaries containing:
+            - entity_group:	The type for the entity being recognized (model specific).
+            - score:    	How likely the entity was recognized.
+            - word:	        The string that was captured.
+            - start:	    The offset stringwise where the answer is located. Useful to disambiguate if word occurs multiple times.
+            - end:	        The offset stringwise where the answer is located. Useful to disambiguate if word occurs multiple times.
+
+        Raises:
+            [`InferenceTimeoutError`]:
+                If the model is unavailable or the request times out.
+            `HTTPError`:
+                If the request fails with an HTTP error status code other than HTTP 503.
+
+        Example:
+        ```py
+        >>> from huggingface_hub import InferenceClient
+        >>> client = InferenceClient()
+        >>> output = client.token_classification("My name is Sarah Jessica Parker but you can call me Jessica")
+        >>> output
+        [{'entity_group': 'PER',
+        'score': 0.9971321225166321,
+        'word': 'Sarah Jessica Parker',
+        'start': 11,
+        'end': 31},
+        {'entity_group': 'PER',
+        'score': 0.9773476123809814,
+        'word': 'Jessica',
+        'start': 52,
+        'end': 59}]
+        ```
+        """
+        payload: Dict[str, Any] = {"inputs": text}
+        if parameters is not None:
+            payload["parameters"] = parameters
+        response = self.post(
+            json=payload,
+            model=model,
+            task="token-classification",
+        )
+        return _bytes_to_dict(response)
```

Comment:
To be adapted either as

```py
        response = self.post(json={"inputs": text}, model=model, task="token-classification")
        return _bytes_to_dict(response)
```

or 

```py
        # convert 'text' to '[text]'
        response = self.post(json={"inputs": [text]}, model=model, task="token-classification")
       # select only first item 
        return _bytes_to_dict(response)[0]
```

I don't know exactly how the server API works so it would have to be tested :)

Reply:
Used your newly implemented `_bytes_to_list`.